tests/testsuite/lints/unused_optional_dependencies.rs

```diff
@@ -157,3 +157,72 @@ warning: unused optional dependency
         )
         .run();
 }
+
+#[cargo_test(nightly, reason = "edition2024 is not stable")]
+fn inactive_weak_optional_dep() {
+    Package::new("dep", "0.1.0").publish();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            cargo-features = ["edition2024"]
+            [package]
+            name = "foo"
+            version = "0.1.0"
+            edition = "2024"
+
+            [dependencies]
+
+            [features]
+            feat = ["dep?/feat"]
```

Comment:
- `dep` doesn't have the `feature` feature
- We should clarify the names so people are less likely to be confused  / make the test output clearer
  - `dep` -> `dep_name`
  - `dep`s `feature` -> `dep_feature`
  - `foo`s `feature -> `foo_feature`